### PR TITLE
[feat]: [CDS-70000]: Added settings update API DTOs for Overrides V2 Feature

### PR DESCRIPTION
**Pull request by lovish1999**
## Describe your changes
Added the DTOs for the API to update user settings to use Overrides V2 Feature

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48221)

### CODE REVIEW

**Comment by lovish1999** [comment]
trigger uts

**Comment by lovish1999** [comment]
trigger messagecheck

**Comment by lovish1999** [comment]
trigger buildnum

**Comment by lovish1999** [comment]
trigger messagecheck

**Comment by lovish1999** [comment]
trigger uts

File: 120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java [hunk 1]
```diff
@@ -336,4 +339,52 @@ public class ServiceOverridesResource {
...
+  public ResponseDTO<ServiceOverrideSettingsUpdateResponseDTO>
+  settingsUpdate(@Parameter(description = NGCommonEntityConstants.ACCOUNT_PARAM_MESSAGE) @NotNull @QueryParam(
+                     NGCommonEntityConstants.ACCOUNT_KEY) String accountId,
+      @Parameter(description = NGCommonEntityConstants.ORG_PARAM_MESSAGE) @QueryParam(
+          NGCommonEntityConstants.ORG_KEY) String orgIdentifier,
+      @Parameter(description = NGCommonEntityConstants.PROJECT_PARAM_MESSAGE) @QueryParam(
+          NGCommonEntityConstants.PROJECT_KEY) String projectIdentifier) {
+    ServiceOverrideSettingsUpdateResponseDTO serviceOverrideSettingsUpdateResponseDTO =
+        serviceOverrideV2SettingsUpdateService.settingsUpdateToV2(
+            accountId, orgIdentifier, projectIdentifier, false, false);
```
Comment (tchaurasiya) [inline 120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java:362]: as we discussed this migrateChildren field should come from query param

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/OverridesGroupSettingsUpdateResult.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+@Data
+@Builder
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(NON_EMPTY)
+@Schema(name = "OverridesGroupSettingsUpdateResult",
+    description = "This is the Scope Level Service Override Yaml Update Response")
+public class OverridesGroupSettingsUpdateResult {
+  long yamlUpdatedServiceOverridesCount;
+  long totalServiceOverrideCount;
```
Comment (tchaurasiya) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/OverridesGroupSettingsUpdateResult.java:29]: we can remove this in next pr

**Review by tchaurasiya — CHANGES_REQUESTED** [review]

**Comment by lovish1999** [comment]
trigger ut6

**Comment by lovish1999** [comment]
trigger ut6

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/OverridesGroupSettingsUpdateResult.java (hunk 2 above)
Comment (lovish1999) [reply 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/OverridesGroupSettingsUpdateResult.java]: ok

File: 120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java (hunk 1 above)
Comment (lovish1999) [reply 120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java]: ok

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/OrgLevelOverrideV2SettingsUpdateResponseDTO.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(NON_EMPTY)
+@Schema(name = "OrgLevelOverrideV2SettingsUpdateResponseDTO",
+    description = "This is the Org Level Override V2 Settings Update Response")
+public class OrgLevelOverrideV2SettingsUpdateResponseDTO {
+  String accountId;
+  String orgIdentifier;
+  boolean isSettingsUpdateSuccessFul;
+  Call<ResponseDTO<List<SettingResponseDTO>>> settingResponseDTO;
```
[the remark this answers is not in the document]
Comment (lovish1999) [reply 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/OrgLevelOverrideV2SettingsUpdateResponseDTO.java]: ok

File: 120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java [hunk 4]
```diff
@@ -336,4 +339,52 @@ public class ServiceOverridesResource {
...
+    return ResponseDTO.newResponse(serviceOverrideSettingsUpdateResponseDTO);
+  }
+
+  @POST
+  @Path("/settingsUpdateScope")
```
[the remark this answers is not in the document]
Comment (lovish1999) [reply 120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java]: ok

**Comment by lovish1999** [comment]
trigger uts

**Comment by lovish1999** [comment]
trigger messagecheck

**Comment by tchaurasiya** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java` line 364 at r2](https://reviewable.io/reviews/harness/harness-core/48221#-NWaAEK41ooti13bTUok:-NWaAEK41ooti13bTUol:bhiiy7h) ([raw file](https://github.com/harness/harness-core/blob/eb7507f7eb419fbc198fb163d519c9a9279ca97c/120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java#L364)):*
> ```Java
>     ServiceOverrideSettingsUpdateResponseDTO serviceOverrideSettingsUpdateResponseDTO =
>         serviceOverrideV2SettingsUpdateService.settingsUpdateToV2(
>             accountId, orgIdentifier, projectIdentifier, migrateChildren, false);
> ```

is making "isRevert" always false intentional?
Shouldn't we also take this as query param

File: 120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java [hunk 5]
```diff
@@ -336,4 +339,29 @@ public class ServiceOverridesResource {
...
+      })
+  public ResponseDTO<ServiceOverrideSettingsUpdateResponseDTO>
+  settingsUpdate(@Parameter(description = NGCommonEntityConstants.ACCOUNT_PARAM_MESSAGE) @NotNull @QueryParam(
+                     NGCommonEntityConstants.ACCOUNT_KEY) String accountId,
+      @Parameter(description = NGCommonEntityConstants.ORG_PARAM_MESSAGE) @QueryParam(
+          NGCommonEntityConstants.ORG_KEY) String orgIdentifier,
+      @Parameter(description = NGCommonEntityConstants.PROJECT_PARAM_MESSAGE) @QueryParam(
+          NGCommonEntityConstants.PROJECT_KEY) String projectIdentifier,
+      @Parameter(description = "Boolean field to decide whether to also update settings of all lower scoped entities.")
+      @QueryParam("migrateChildren") Boolean migrateChildren) {
```
Comment (tchaurasiya) [inline 120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java:361]: lets use primitive boolean

**Review by tchaurasiya — CHANGES_REQUESTED** [review]

**Review by lovish1999 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48221)* status: 8 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java` line 361 at r2](https://reviewable.io/reviews/harness/harness-core/48221#gh-1208963295:-NWaIwyh93kHIfFkOWE4:b5mpjzs) ([raw file](https://github.com/harness/harness-core/blob/eb7507f7eb419fbc198fb163d519c9a9279ca97c/120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java#L361)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

lets use primitive boolean
</blockquote></details>

Okay, but what will be the issue with this one ? 
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java` line 364 at r2](https://reviewable.io/reviews/harness/harness-core/48221#-NWaAEK41ooti13bTUok:-NWaIPbO4IE9BjY0Ep8Y:b52yl66) ([raw file](https://github.com/harness/harness-core/blob/eb7507f7eb419fbc198fb163d519c9a9279ca97c/120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java#L364)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

is making "isRevert" always false intentional?
Shouldn't we also take this as query param
</blockquote></details>

We decided to have settings revert as part of the revert API call itself. So this API call will only have isRevert=false. And the revert = true part will be added as part of the revert API call. Although, I also prefer to not go with this approach and instead keep isRevert as boolean in settings update API.

**Comment by tchaurasiya** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java` line 364 at r2](https://reviewable.io/reviews/harness/harness-core/48221#-NWaAEK41ooti13bTUok:-NWarB3e8NvFJPb264WI:bnkem2l) ([raw file](https://github.com/harness/harness-core/blob/eb7507f7eb419fbc198fb163d519c9a9279ca97c/120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java#L364)):*
<details><summary><i>Previously, lovish1999 (Lovish Bansal) wrote…</i></summary><blockquote>

We decided to have settings revert as part of the revert API call itself. So this API call will only have isRevert=false. And the revert = true part will be added as part of the revert API call. Although, I also prefer to not go with this approach and instead keep isRevert as boolean in settings update API. 
</blockquote></details>

yeah make sense!

**Comment by tchaurasiya** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java` line 361 at r2](https://reviewable.io/reviews/harness/harness-core/48221#gh-1208963295:-NWas19v099lY-9oVLNv:b26gb7i) ([raw file](https://github.com/harness/harness-core/blob/eb7507f7eb419fbc198fb163d519c9a9279ca97c/120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java#L361)):*
<details><summary><i>Previously, lovish1999 (Lovish Bansal) wrote…</i></summary><blockquote>

Okay, but what will be the issue with this one ? 
</blockquote></details>

We should not use Boolean class if very specific case are not needed. (like Collections and generic types in Java cannot directly use primitive types as type parameters)

pros of primitive
*more efficient in terms of memory usage and performance compared
* we dont need to handle Nullability

**Review by lovish1999 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48221)* status: Some of the checks are too old. 8 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java` line 361 at r2](https://reviewable.io/reviews/harness/harness-core/48221#gh-1208963295:-NWcHiIg4OKQ8sbc05VR:bdcrwo8) ([raw file](https://github.com/harness/harness-core/blob/eb7507f7eb419fbc198fb163d519c9a9279ca97c/120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java#L361)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

We should not use Boolean class if very specific case are not needed. (like Collections and generic types in Java cannot directly use primitive types as type parameters)

pros of primitive
*more efficient in terms of memory usage and performance compared
* we dont need to handle Nullability
</blockquote></details>

Okay, thanks. 
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java` line 364 at r2](https://reviewable.io/reviews/harness/harness-core/48221#-NWaAEK41ooti13bTUok:-NWcHdbYCWOVjZi7zDL6:b-y071is) ([raw file](https://github.com/harness/harness-core/blob/eb7507f7eb419fbc198fb163d519c9a9279ca97c/120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java#L364)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

yeah make sense!
</blockquote></details>

So, then should we have this change ?

**Comment by tchaurasiya** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java` line 364 at r2](https://reviewable.io/reviews/harness/harness-core/48221#-NWaAEK41ooti13bTUok:-NWdDyeMEFxICJKMZV55:by7clr) ([raw file](https://github.com/harness/harness-core/blob/eb7507f7eb419fbc198fb163d519c9a9279ca97c/120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java#L364)):*
<details><summary><i>Previously, lovish1999 (Lovish Bansal) wrote…</i></summary><blockquote>

So, then should we have this change ?
</blockquote></details>

No. we should go ahead, what you already have implemented.

**Review by tchaurasiya — COMMENTED** [review]
> Reviewed 2 of 8 files at r1, 3 of 6 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48221)* status: Some of the checks are too old. 8 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained.

**Review by tchaurasiya — APPROVED** [review]
> Reviewed 3 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48221)* status: Some of the checks are too old. 8 file(s) to review. 1 of 2 approvals obtained.

**Comment by lovish1999** [comment]
trigger review

**Pull request merged**